deeplake/core/transform/transform_tensor.py

```diff
@@ -12,9 +14,40 @@ def __init__(self, name, dataset, items=None, slice_list=None) -> None:
         self.slice_list = slice_list or []
         self.length = None
         self._ndim = None
+        if not self.items:
+            self.items.append(True)
```

Comment:
TransformTensor can store items in 2 formats:

* Numpy only: each item in items list is a numpy array. The numpy arrays are actually batches of samples
* Not numpy only: Each item in items list is a sample

The idea here is to not break up numpy arrays whenever possible. This has 2 benefits: 1 - extend optimizations, 2 - converting numpy arrays to lists creates a lot of python objects-> more memory hog + GC overhead.